src/openrct2/world/Sprite.h

```diff
@@ -275,209 +255,6 @@ rct_sprite_checksum sprite_checksum();
-
-        while (iter != end && Entity == nullptr)
-        {
-            Entity = GetEntity<T>(*iter++);
-        }
-        return *this;
-    }
-
-    EntityTileIterator operator++(int)
-    {
-        EntityTileIterator retval = *this;
-        ++(*this);
-        return *iter;
-    }
-    bool operator==(EntityTileIterator other) const
-    {
-        return Entity == other.Entity;
-    }
-    bool operator!=(EntityTileIterator other) const
-    {
-        return !(*this == other);
-    }
-    T* operator*()
-    {
-        return Entity;
-    }
-    // iterator traits
-    using difference_type = std::ptrdiff_t;
-    using value_type = T;
-    using pointer = const T*;
-    using reference = const T&;
-    using iterator_category = std::forward_iterator_tag;
-};
-
-template<typename T = SpriteBase> class EntityTileList
-{
-private:
-    const std::vector<uint16_t>& vec;
-
-public:
-    EntityTileList(const CoordsXY& loc)
-        : vec(GetEntityTileList(loc))
-    {
-    }
-
-    EntityTileIterator<T> begin()
-    {
-        return EntityTileIterator<T>(std::begin(vec), std::end(vec));
-    }
-    EntityTileIterator<T> end()
-    {
-        return EntityTileIterator<T>(std::end(vec), std::end(vec));
-    }
-};
-
-template<typename T> class EntityListIterator
-{
-private:
-    std::list<uint16_t>::const_iterator iter;
-    std::list<uint16_t>::const_iterator end;
-    T* Entity = nullptr;
-
-public:
-    EntityListIterator(std::list<uint16_t>::const_iterator _iter, std::list<uint16_t>::const_iterator _end)
-        : iter(_iter)
-        , end(_end)
-    {
-        ++(*this);
-    }
-    EntityListIterator& operator++()
-    {
-        Entity = nullptr;
-
-        while (iter != end && Entity == nullptr)
-        {
-            Entity = GetEntity<T>(*iter++);
-        }
-        return *this;
-    }
-
-    EntityListIterator operator++(int)
-    {
-        EntityListIterator retval = *this;
-        ++(*this);
-        return *iter;
-    }
-    bool operator==(EntityListIterator other) const
-    {
-        return Entity == other.Entity;
-    }
-    bool operator!=(EntityListIterator other) const
-    {
-        return !(*this == other);
-    }
-    T* operator*()
-    {
-        return Entity;
-    }
-    // iterator traits
-    using difference_type = std::ptrdiff_t;
-    using value_type = T;
-    using pointer = const T*;
-    using reference = const T&;
-    using iterator_category = std::forward_iterator_tag;
-};
-
-template<typename T = SpriteBase> class EntityList
-{
-private:
-    using EntityListIterator_t = EntityListIterator<T>;
-    const std::list<uint16_t>& vec;
-
-public:
-    EntityList(EntityListId type)
-        : vec(GetEntityList(type))
-    {
-    }
-
-    EntityListIterator_t begin()
-    {
-        return EntityListIterator_t(std::cbegin(vec), std::cend(vec));
-    }
-    EntityListIterator_t end()
-    {
-        return EntityListIterator_t(std::cend(vec), std::cend(vec));
-    }
-};
-
 class EntityTweener
```

Comment:
Would suggest moving the entity tweener into a separate file as well. It's used in perhaps two, three places

Reply:
Not worth it. `Sprite.h` is now used only in `.cpp` files and three scripting includes, all of which seem to require full `Sprite.h` anyway